src/Operations.jl

```diff
@@ -1782,4 +1797,114 @@ function status(env::EnvCache, pkgs::Vector{PackageSpec}=PackageSpec[];
+        uuid_list;
+        target_name,
+        allow_earlier_backwards_compatible_versions,
+    )
+    return all(results)
+end
+
+function check_force_latest_compatible_version(ctx::Types.Context,
+                                               uuid::Base.UUID;
+                                               target_name= nothing,
+                                               allow_earlier_backwards_compatible_versions::Bool = true)
+    dep = ctx.env.manifest[uuid]
+    name = dep.name
+    active_version = dep.version
+    has_compat = haskey(ctx.env.project.compat, name)
+    if !has_compat
+        if name != target_name
+            @warn(
+                "Package does not have a [compat] entry",
+                name, uuid, active_version, target_name,
+            )
+        end
+        return true
+    end
+    compat_entry_string = ctx.env.project.compat[name]
+    latest_compatible_version = get_latest_compatible_version(
+        ctx,
+        uuid,
+        compat_entry_string,
+    )
+    earliest_backwards_compatible_version = get_earliest_backwards_compatible_version(latest_compatible_version)
+    if allow_earlier_backwards_compatible_versions
+        result = active_version >= earliest_backwards_compatible_version
+    else
+        result = active_version >= latest_compatible_version
+    end
+    if !result
+        @error(
+            "Package is not at the latest compatible version",
+            name,
+            uuid,
+            compat_entry_string,
+            active_version,
+            latest_compatible_version,
+            earliest_backwards_compatible_version,
+            allow_earlier_backwards_compatible_versions,
+        )
+    end
+    return result
+end
+
+function get_earliest_backwards_compatible_version(ver::Base.VersionNumber)
+    major = ver.major
+    minor = ver.minor
+    patch = ver.patch
+    (major > 0) && return Base.VersionNumber(major, 0, 0)
+    (minor > 0) && return Base.VersionNumber(0, minor, 0)
+    return Base.VersionNumber(0, 0, patch)
+end
+
+function get_latest_compatible_version(ctx::Types.Context,
+                                       uuid::Base.UUID,
+                                       compat_entry_string::AbstractString)
+    all_registered_versions = get_all_registered_versions(ctx, uuid)
+    compat_spec = Pkg.Types.semver_spec(compat_entry_string)
+    compatible_versions = filter(
+        x -> x in compat_spec,
+        all_registered_versions,
+    )
+    latest_compatible_version = maximum(compatible_versions)
+    return latest_compatible_version
+end
+
+function get_all_registered_versions(ctx::Types.Context,
+                                     uuid::Base.UUID)
+    versions = Set{VersionNumber}()
+    for reg in ctx.registries
+        pkg = get(reg, uuid, nothing)
+        if pkg !== nothing
+            info = Registry.registry_info(pkg)
+            for v in keys(info.version_info)
+                push!(versions, v)
+            end
```

Comment:
I think this can be stated as:
```suggestion
            union!(versions, keys(info.version_info))
```